packages/patterns/src/isPatternEqual.ts

```diff
@@ -19,8 +19,13 @@
  */
 import convertToCSS from './convertToCSS';
 import generatePatternStyles from './generatePatternStyles';
+import type { Pattern } from './types';
 
-export default function isPatternEqual(p1, p2, patternType = undefined) {
+export default function isPatternEqual(
+  p1: Pattern,
+  p2: Pattern,
+  patternType?: string
```

Comment:
These are the only valid values AFAIK. Could be moved to `types.ts` as well.

```suggestion
  patternType?: 'color' | 'background'
```

Reply:
Isn't it used for `border-color` too? But yes, it makes sense to narrow this down as much as possible. I'll look into it.

Reply:
It is also used for `fill` and `stroke`.

But there are also other cases making narrowing tricky. [Here](https://github.com/GoogleForCreators/web-stories-wp/blob/512328ada20ed77687817ee479bd09255836ca61/packages/element-library/src/product/display.js#L132) it is used for a CSS variable (`--box-shadow-color`), which isn't an unreasonable usage, and the naming is completely arbitrary.

We could enforce a certain number of properties plus any string that starts with `'--'`, but that seems like overkill?